### PR TITLE
fixes cloning an empty repo

### PR DESCRIPTION
**Pull request by na4ma4**
As per go-git/go-git#118, it's expected behaviour to just fail to clone an empty repo, this will on receiving the empty repository error, init, add remote and add branch tracking remote to the local repository.

### CODE REVIEW

File: cmd/grit/clone.go [hunk 1]
```diff
@@ -38,6 +40,23 @@ func clone(cfg grit.Config, idx *index.Index, c *cli.Context) error {
 	case transport.ErrEmptyRemoteRepository:
 		fmt.Fprintln(os.Stderr, "cloned an empty repository")
```
Comment (jmalloc) [inline cmd/grit/clone.go]: This should probably only be shown after the init, etc works successfully.

File: cmd/grit/clone.go [hunk 2]
```diff
@@ -38,6 +40,23 @@ func clone(cfg grit.Config, idx *index.Index, c *cli.Context) error {
...
 
+		r, err := git.PlainInit(dir, false /* isBare */)
+		if err != nil {
+			return err
+		}
+
```
Comment (jmalloc) [inline cmd/grit/clone.go:46]: There needs to be something that removes `dir` if creating the remote or the default branch fail.

File: etc/complete-example.toml [hunk 3]
```diff
@@ -26,6 +26,12 @@
 # Will clone git@github.com:jmalloc/grit.git to ~/grit/github.com/jmalloc/grit
 root = "~/my-repos"
 
+# "default-branch" is is used when cloning an empty repo.
+
```
Comment (jmalloc) [inline etc/complete-example.toml]:
```suggestion
# "default-branch" is is used when cloning an empty repo.
#
```

**Review by jmalloc — CHANGES_REQUESTED** [review]

File: cmd/grit/clone.go (hunk 2 above)
Comment (na4ma4) [reply cmd/grit/clone.go:46]: Added the `os.RemoveAll(dir)` in

**Review by na4ma4 — COMMENTED** [review]
That should cover all the changes.

Directory is removed if any step of the Init/Add Origin/Add Branch fails.

**Pull request merged**